code/game/objects/items/weapons/tools/screwdriver.dm

```diff
@@ -16,24 +16,33 @@
 	attack_verb = list("stabbed")
 	lock_picking_level = 5
 	sharp = TRUE
+	applies_material_colour = TRUE
 
 	var/build_from_parts = TRUE
 	var/valid_colours = list(COLOR_RED, COLOR_CYAN_BLUE, COLOR_PURPLE, COLOR_CHESTNUT, COLOR_GREEN, COLOR_TEAL, COLOR_ASSEMBLY_YELLOW, COLOR_BOTTLE_GREEN, COLOR_VIOLET, COLOR_GRAY80, COLOR_GRAY20)
+	var/handle_color
 
 /obj/item/screwdriver/Initialize()
-	if(build_from_parts)
-		icon_state = "screwdriver_handle"
-		color = pick(valid_colours)
-		overlays += overlay_image(icon, "screwdriver_hardware", flags=RESET_COLOR)
 	if (prob(75))
 		src.pixel_y = rand(0, 16)
 	. = ..()
 
+/obj/item/screwdriver/on_update_icon()
+	..()
+	if(build_from_parts)
+		icon_state = "hardware"
+		if(!handle_color)
+			handle_color = pick(valid_colours)
+		overlays += overlay_image(icon, "handle", handle_color, flags=RESET_COLOR)
```

Comment:
While you are doing this it might be worth switching it over to use add_overlay, unless /obj/item or /obj do something with manual overlays.

Reply:
/obj/item indoot do overlays.Cut() in base update icon proc